sdk/test-utils/perfstress/src/testProxyHttpClient.ts

```diff
@@ -0,0 +1,209 @@
+      redirectedUrl.setHost(providedUrl.getHost());
+      redirectedUrl.setScheme(providedUrl.getScheme());
+      redirectedUrl.setPort(providedUrl.getPort());
+      upstreamUrl.setPath(undefined);
+      request.headers.set("x-recording-upstream-base-uri", upstreamUrl.toString());
+      request.url = redirectedUrl.toString();
+    }
+
+    return await super.sendRequest(request);
+  }
+
+  async startRecording(): Promise<void> {
+    const startUri = this._uri + paths.record + paths.start;
+    const req = this._createRecordingRequest(startUri);
+    const rsp = await this._httpClient.sendRequest(req);
+    if (rsp.status !== 200) {
+      throw new Error("Start request failed.");
+    }
+    const id = rsp.headers.get("x-recording-id");
+    if (!id) {
+      throw new Error("No recording ID returned.");
+    }
+    this._recordingId = id;
+  }
+
+  async stopRecording(): Promise<void> {
+    const stopUri = this._uri + paths.record + paths.stop;
+    const req = this._createRecordingRequest(stopUri);
+    req.headers.set("x-recording-id", this._recordingId!);
+    await this._httpClient.sendRequest(req);
+  }
+
+  async startPlayback(): Promise<void> {
+    const startUri = this._uri + paths.playback + paths.start;
+    const req = this._createRecordingRequest(startUri);
+    req.headers.set("x-recording-id", this._recordingId!);
+    const rsp = await this._httpClient.sendRequest(req);
+    if (rsp.status !== 200) {
+      throw new Error("Start request failed.");
+    }
+    const id = rsp.headers.get("x-recording-id");
+    if (!id) {
+      throw new Error("No recording ID returned.");
+    }
+    this._recordingId = id;
+  }
+
+  async stopPlayback(): Promise<void> {
+    const stopUri = this._uri + paths.playback + paths.stop;
+    const req = this._createRecordingRequest(stopUri);
+    req.headers.set("x-recording-id", this._recordingId!);
+    req.headers.set("x-purge-inmemory-recording", "true");
+    await this._httpClient.sendRequest(req);
+
+    this._mode = "live";
+    this._recordingId = undefined;
+  }
+
+  private _createRecordingRequest(uri: string) {
+    const req = new WebResource(uri, "POST");
+    if (this._recordingId !== undefined) {
+      req.headers.set("x-recording-id", this._recordingId);
+    }
+    return req;
+  }
+}
+
+export class TestProxyHttpClient {
+  private _uri: string;
+  private _httpClient: HttpClient;
+  private _recordingId?: string;
+  public _mode!: string;
+
+  constructor(uri: string) {
+    this._uri = uri;
+    this._httpClient = createDefaultHttpClient();
+  }
+
+  async sendRequest(request: PipelineRequest): Promise<PipelineRequest> {
+    if (this._recordingId && (this._mode === "record" || this._mode === "playback")) {
+      request.headers.set("x-recording-id", this._recordingId);
+      request.headers.set("x-recording-mode", this._mode);
+      request.headers.set("x-recording-remove", "false");
+
+      const upstreamUrl = URLBuilder.parse(request.url);
```

Comment:
Do we really want to depend on `URLBuilder` for this logic? It's notoriously bad at handling URLs consistently, and will eventually go away along with `core-http`

Reply:
I see.
Any recommendations on what to use?

Reply:
new URL(url) should work, I think

Reply:
Okay, will use that.

Reply:
Updated to URL